cpp/src/io/parquet/page_data.cu

```diff
@@ -353,11 +367,36 @@ __device__ int gpuDecodeDictionaryIndices(volatile page_state_s* s, int target_p
           dict_idx &= (1 << dict_bits) - 1;
         }
       }
-      s->dict_idx[(pos + t) & (non_zero_buffer_size - 1)] = dict_idx;
+
+      // if we're not computing sizes, store off the dictionary index
+      if constexpr (!sizes_only) { s->dict_idx[(pos + t) & (non_zero_buffer_size - 1)] = dict_idx; }
+    }
+
+    // if we're computing sizes, add the length(s)
+    if constexpr (sizes_only) {
+      int const len = [&]() {
+        if (t >= batch_len) { return 0; }
+        // we may end up decoding more indices than we asked for. so don't include those in the
+        // size calculation
+        if (pos + t >= target_pos) { return 0; }
+        // TODO:  refactor this with gpuGetStringData / gpuGetStringSize
```

Comment:
This could probably be refactored at this point. Would it be ok to do this as a followup (in the interest of not destabilizing things at this late stage)?

Reply:
https://github.com/rapidsai/cudf/issues/12143

Reply:
Sure! I brought this up mainly because it can be used in my binary decimal work as well. Glad you've added those handy functions!